### PR TITLE
feat: tonality DIN45681

### PR DESCRIPTION
**Pull request by ansaminard**
New helper class for DIN45681
Associated test class
New entries for doc

Unrelated:
Removed a useless line in SWL

### CODE REVIEW

**Review by a-bouth — DISMISSED** [review]

File: src/ansys/sound/core/psychoacoustics/tonality_din_45681.py [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+    @signal.setter
+    def signal(self, signal: Field):
+        """Set signal property."""
+        if not (isinstance(signal, Field) or signal is None):
+            raise PyAnsysSoundException("Signal must be specified as a DPF field.")
+        self.__signal = signal
+
+    @property
+    def window_length(self) -> float:
+        """Analysis window length in s. Default is 3.0  s."""
```
Comment (mattbennett72) [inline src/ansys/sound/core/psychoacoustics/tonality_din_45681.py:89]: extra space here between 3.0 and s

File: src/ansys/sound/core/psychoacoustics/tonality_din_45681.py [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+        """
+        return self.get_output_as_nparray()[6]
+
+    def get_time_scale(self) -> npt.ArrayLike:
+        """Get the DIN 45681 time scale in s.
+
+        Returns
+        -------
+        numpy.ndarray
+            Computation timestamps in s.
```
Comment (mattbennett72) [inline src/ansys/sound/core/psychoacoustics/tonality_din_45681.py:305]: Just checking that you intended to use 'timestamps' here.

File: src/ansys/sound/core/psychoacoustics/tonality_din_45681.py [hunk 3]
```diff
@@ -0,0 +1,487 @@
...
+        """
+        return self.get_output_as_nparray()[7]
+
+    def get_spectrum_number(self) -> int:
+        """Get the number of spectra.
+
+        Returns the number of analyzed spectra, ie the number of computation time steps.
```
Comment (mattbennett72) [inline src/ansys/sound/core/psychoacoustics/tonality_din_45681.py:312]: we are encouraged not to use latin abbreviations in documentation (not universally understood). 'that is' is better.

File: src/ansys/sound/core/psychoacoustics/tonality_din_45681.py [hunk 4]
```diff
@@ -0,0 +1,487 @@
...
+        int
+            Number of spectra.
+        """
+        return len(self.get_output()[3])
+
+    def get_spectrum_details(self, spectrum_index: int = 0) -> tuple[float, float, float]:
+        """Get the spectrum data.
+
+        Returns the data (decisive difference, uncertainty, and decisive frequency) corresponding
+        to a specific spectrum (ie time step).
```
Comment (mattbennett72) [inline src/ansys/sound/core/psychoacoustics/tonality_din_45681.py:325]: same note on i.e. as above

File: src/ansys/sound/core/psychoacoustics/tonality_din_45681.py [hunk 5]
```diff
@@ -0,0 +1,487 @@
...
+        return (
+            self.get_output_as_nparray()[3][spectrum_index],
+            self.get_output_as_nparray()[4][spectrum_index],
+            self.get_output_as_nparray()[5][spectrum_index],
+        )
+
+    def get_tone_number(self, spectrum_index: int = 0) -> int:
+        """Get the number of tones.
+
+        Returns the number of tones detected in a specific spectrum (ie time step).
```
Comment (mattbennett72) [inline src/ansys/sound/core/psychoacoustics/tonality_din_45681.py:353]: (that is, time step)

File: src/ansys/sound/core/psychoacoustics/tonality_din_45681.py [hunk 6]
```diff
@@ -0,0 +1,487 @@
...
+        spectrum = collection.get_entry(spectrum_index)
+
+        return len(spectrum.get_property("differences"))
+
+    def get_tone_details(
+        self, spectrum_index: int = 0, tone_index: int = 0
+    ) -> tuple[float, float, float, str, float, float, float, float, float, float]:
+        """Get the tone data.
+
+        Returns all data corresponding to a specific detected tone, in a specific spectrum (ie time
```
Comment (mattbennett72) [inline src/ansys/sound/core/psychoacoustics/tonality_din_45681.py:375]: (that is, time step)

File: src/ansys/sound/core/psychoacoustics/tonality_din_45681.py (hunk 2 above)
Comment (ansaminard) [reply src/ansys/sound/core/psychoacoustics/tonality_din_45681.py:305]: Yes I did. I shouldn't?

Comment (anssmolla) [reply src/ansys/sound/core/psychoacoustics/tonality_din_45681.py:310]: maybe the term "timestamp" is used in really specific cases? What means timestamp here @ansaminard?

**Review by anssmolla — DISMISSED** [review]

**Review by anshlachamb — DISMISSED** [review]

File: src/ansys/sound/core/psychoacoustics/tonality_din_45681.py (hunk 2 above)
Comment (anssmolla) [reply src/ansys/sound/core/psychoacoustics/tonality_din_45681.py:310]: @ansaminard maybe we could try somthing like:

Vector of the time, in second, corresponding to the vectors [list of the data vector to which this time applies]"

what do you think?

Comment (ansaminard) [reply src/ansys/sound/core/psychoacoustics/tonality_din_45681.py:310]: @anssmolla I'm still failing to see why "timestamps" is wrong here...

**Review by a-bouth — DISMISSED** [review]

File: src/ansys/sound/core/psychoacoustics/tonality_din_45681.py (hunk 2 above)
Comment (ansaminard) [reply src/ansys/sound/core/psychoacoustics/tonality_din_45681.py]: ok done

**Review by a-bouth — APPROVED** [review]

**Review by anshlachamb — APPROVED** [review]

**Review by anssmolla — APPROVED** [review]

**Pull request merged**